tools/src/lib.rs

```diff
@@ -839,15 +848,46 @@ where
         .into_iter()
         .max_by(|a, b| W::partial_cmp(a, b).unwrap())
         .unwrap();
-    let zoom_factor = (O::max_value() - O::from(inputs.len()).unwrap())
-        .to_f64()
-        .unwrap()
-        / sum.to_f64().unwrap();
-    let _1 = O::one();
+    let (min_input, max_input) = inputs.clone().flatten().minmax().into_option().unwrap();
+    if min_input == max_input {
+        return vec![O::one(); count * criterion_count];
+    }
```

Comment:
Actually it might be better to have `zoom_in` return `None` when `min = max`. Then we can leave the weight/adjwgt parameter of scotch/metis as the default.

I'll upload a new patch soon.

Reply:
done